src/simulation.rs

```diff
@@ -141,6 +142,45 @@ impl Simulation {
         // wait for report response
         let final_report = events_report.await.expect("events collection to end");
 
+        let ready_users = self.entities.values().filter_map(|entity| {
```

Comment:
Let's extract this to a separate function to avoid having too many details in the `Simulation#run`.